prebuild/emscripten/post.js

```diff
@@ -0,0 +1,1010 @@
+  const [_msg, msg_len] = ensureRef(msg);
+  const [_aug, aug_len] = ensureRef(aug);
+
+  return wrapPointer(_emscripten_bind_P2_hash_to_5(self, _msg, msg_len, _dst, _aug, aug_len), P2);
+};;
+
+P2.prototype['encode_to'] = P2.prototype.encode_to = /** @suppress {undefinedVars, duplicate} @this{Object} */function(msg, DST, aug) {
+  var self = this.ptr;
+  ensureCache.prepare();
+
+  const [_dst] = ensureRef(DST);
+  const [_msg, msg_len] = ensureRef(msg);
+  const [_aug, aug_len] = ensureRef(aug);
+
+  return wrapPointer(_emscripten_bind_P2_encode_to_5(self, _msg, msg_len, _dst, _aug, aug_len), P2);
+};;
+
+P2.prototype['mult'] = P2.prototype.mult = /** @suppress {undefinedVars, duplicate} @this{Object} */function(scalar) {
+  var self = this.ptr;
+  const [_scalar, scalar_len] = ensureRef(scalar);
+  return wrapPointer(_emscripten_bind_P2_mult_1(self, _scalar, scalar_len * 8), P2);
+};;
+
+P2.prototype['neg'] = P2.prototype.neg = /** @suppress {undefinedVars, duplicate} @this{Object} */function() {
+  var self = this.ptr;
+  return wrapPointer(_emscripten_bind_P2_neg_0(self), P2);
+};;
+
+P2.prototype['cneg'] = P2.prototype.cneg = /** @suppress {undefinedVars, duplicate} @this{Object} */function(flag) {
+  var self = this.ptr;
+  if (flag && typeof flag === 'object') flag = flag.ptr;
+  return wrapPointer(_emscripten_bind_P2_cneg_1(self, flag), P2);
+};;
+
+P2.prototype['add'] = P2.prototype.add = /** @suppress {undefinedVars, duplicate} @this{Object} */function(a) {
+  var self = this.ptr;
+  switch(a.constructor) {
+    case P2:
+      return wrapPointer(_emscripten_bind_P2_add_1__P2(self, a.ptr), P2);
+    case P2_Affine:
+      return wrapPointer(_emscripten_bind_P2_add_1__P2_Affine(self, a.ptr), P2);
+    default:
+      throw new Error("unsuported type for `a` arg");
+  }
+};;
+
+P2.prototype['dbl'] = P2.prototype.dbl = /** @suppress {undefinedVars, duplicate} @this{Object} */function() {
+  var self = this.ptr;
+  return wrapPointer(_emscripten_bind_P2_dbl_0(self), P2);
+};;
+
+  P2.prototype['__destroy__'] = P2.prototype.__destroy__ = /** @suppress {undefinedVars, duplicate} @this{Object} */function() {
+  var self = this.ptr;
+  _emscripten_bind_P2___destroy___0(self);
+};
+// SecretKey
+/** @suppress {undefinedVars, duplicate} @this{Object} */function SecretKey() {
+  this.ptr = _emscripten_bind_SecretKey_SecretKey_0();
+  getCache(SecretKey)[this.ptr] = this;
+};;
+SecretKey.prototype = Object.create(WrapperObject.prototype);
+SecretKey.prototype.constructor = SecretKey;
+SecretKey.prototype.__class__ = SecretKey;
+SecretKey.__cache__ = {};
+Module['SecretKey'] = SecretKey;
+
+SecretKey.prototype['keygen'] = SecretKey.prototype.keygen = /** @suppress {undefinedVars, duplicate} @this{Object} */function(ikm, info) {
+  var self = this.ptr;
+  var ikm_len = 0;
+  ensureCache.prepare();
+  if (ikm && typeof ikm === 'object') {
+    if (ikm instanceof Uint8Array) {
+      ikm_len = ikm.length;
+      ikm = ensureUint8(ikm, ikm.length);
```

Comment:
`ensureUint8` doesn't take length. So passing `ikm.length` is redundant, right? What might be appropriate is to double-check that the length is not shorter than 32.